cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -611,10 +621,31 @@ func (c *machineController) listScalableResources() ([]*unstructured.Unstructure
 }
 
 func (c *machineController) listResources(lister cache.GenericLister) ([]*unstructured.Unstructured, error) {
-	return listResources(lister.ByNamespace(metav1.NamespaceAll), labels.Everything())
+	if len(c.autoDiscoverySpecs) == 0 {
+		return listResources(lister.ByNamespace(metav1.NamespaceAll), "", labels.Everything())
+	}
+
+	var results []*unstructured.Unstructured
+	tracker := map[string]bool{}
+	for _, spec := range c.autoDiscoverySpecs {
+		resources, err := listResources(lister.ByNamespace(spec.namespace), spec.clusterName, spec.labelSelector)
```

Comment:
spec.namespace is a string, so it will default to "" when not set, which is equivalent to all namespaces when passed to a lister. For additional context, if one uses `metav1.NamespaceAll`, it is defined as the empty string: https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#pkg-constants